sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -120,6 +140,123 @@ private static DoFnSignature parseSignature(Class<? extends DoFn> fnClass) {
               errors.forMethod(DoFn.Teardown.class, teardownMethod), teardownMethod));
     }
 
+    DoFnSignature.GetInitialRestrictionMethod getInitialRestriction = null;
+    ErrorReporter getInitialRestrictionErrors = null;
+    if (getInitialRestrictionMethod != null) {
+      getInitialRestrictionErrors =
+          errors.forMethod(DoFn.GetInitialRestriction.class, getInitialRestrictionMethod);
+      builder.setGetInitialRestriction(
+          getInitialRestriction =
+              analyzeGetInitialRestrictionMethod(
+                  getInitialRestrictionErrors, fnToken, getInitialRestrictionMethod, inputT));
+    }
+
+    DoFnSignature.SplitRestrictionMethod splitRestriction = null;
+    if (splitRestrictionMethod != null) {
+      ErrorReporter splitRestrictionErrors =
+          errors.forMethod(DoFn.SplitRestriction.class, splitRestrictionMethod);
+      builder.setSplitRestriction(
+          splitRestriction =
+              analyzeSplitRestrictionMethod(
+                  splitRestrictionErrors, fnToken, splitRestrictionMethod, inputT));
+    }
+
+    DoFnSignature.GetRestrictionCoderMethod getRestrictionCoder = null;
+    if (getRestrictionCoderMethod != null) {
+      ErrorReporter getRestrictionCoderErrors =
+          errors.forMethod(DoFn.GetRestrictionCoder.class, getRestrictionCoderMethod);
+      builder.setGetRestrictionCoder(
+          getRestrictionCoder =
+              analyzeGetRestrictionCoderMethod(
+                  getRestrictionCoderErrors, fnToken, getRestrictionCoderMethod));
+    }
+
+    DoFnSignature.NewTrackerMethod newTracker = null;
+    if (newTrackerMethod != null) {
+      ErrorReporter newTrackerErrors = errors.forMethod(DoFn.NewTracker.class, newTrackerMethod);
+      builder.setNewTracker(
+          newTracker = analyzeNewTrackerMethod(newTrackerErrors, fnToken, newTrackerMethod));
+    }
+
+    // Additional validation for splittable DoFn's.
+    if (processElement.isSplittable()) {
+      if (isBounded == null) {
+        isBounded =
+            processElement.hasReturnValue()
+                ? PCollection.IsBounded.UNBOUNDED
+                : PCollection.IsBounded.BOUNDED;
+      }
+      List<String> missingRequiredMethods = new ArrayList<>();
+      if (getInitialRestriction == null) {
+        missingRequiredMethods.add("@GetInitialRestriction");
+      }
+      if (newTracker == null) {
+        missingRequiredMethods.add("@NewTracker");
+      }
+      // @SplitRestriction and @GetRestrictionCoder are optional.
+      if (!missingRequiredMethods.isEmpty()) {
+        processElementErrors.throwIllegalArgument(
+            "Splittable, but does not define the following required methods: %s",
+            missingRequiredMethods);
+      }
+      processElementErrors.checkArgument(
+          processElement.trackerT().equals(newTracker.trackerT()),
+          "Has tracker type %s, but @NewTracker method %s uses tracker type %s",
```

Comment:
Same comment regarding `@%s` + `NewTracker.class.getSimpleName()` here and throughout (for forbidden methods, required methods, etc.)